src/NuGetForUnity/Editor/PackageSource/NugetApiClientV3.cs

```diff
@@ -45,7 +45,13 @@ internal sealed class NugetApiClientV3 : IDisposable
         [NonSerialized]
         [NotNull]
         private readonly HttpClient httpClient = new HttpClient(
-            new HttpClientHandler { AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate });
+            new HttpClientHandler
+            {
+                AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate,
+#if UNITY_EDITOR_WIN
```

Comment:
Can we enable it on all platforms?
We can't use `#if` toggles because the plugin is also released as a pre-build `.dll` so all compiler directives are already evaluated. If we need this to be platform specific we need to use e.g. `if (Application.platform == RuntimePlatform.WindowsEditor)`

Reply:
@JoC0de 

Thanks for pointing out about the pre-build .dll behaviour. This really helps me.
Upon regression testing, I found that the fix introduces negative effects on MacOS platform. 
So the fix cannot be enabled on all platforms. This is Windows only fix/workaround.

I've updated and moved the HttpClient initialization code to the constructor.
Modified to use Applicaton.platform conditional check as you advised.

Test method:

* Open NuGetForUnity window
* Search NuGet package with some dependencies
* Install NuGet package
* Unintall NuGet package

Test results:

1. DLL version

| Network Environment | Windows | MacOS |
| - | - | - |
| Non Proxy | 20240210_OK | 20240210_OK |
| Proxy | 20240210_OK | 20240210_OK |

2. Script version

| Network Environment | Windows | MacOS |
| - | - | - |
| Non Proxy | 20240210_OK | 20240210_OK |
| Proxy | 20240210_OK | 20240210_OK |

Reply:
Ok thanks for the effort of testing it on the devices.